common/include/pcl/common/impl/bivariate_polynomial.hpp

```diff
@@ -289,6 +289,8 @@ pcl::BivariatePolynomialT<real>::readBinary (std::istream& os)
 {
   memoryCleanUp ();
   os.read (reinterpret_cast<char*> (&this->degree), sizeof (int));
+  PCL_CHECK_IO_STREAM(os, "degree");
+
   unsigned int paramCnt = getNoOfParametersFromDegree (this->degree);
   parameters = new real[paramCnt];
   os.read (reinterpret_cast<char*> (&(*this->parameters)), paramCnt * sizeof (real));
```

Comment:
A check after this read too.

Maybe one check after every read. Since it's a macro, that should be relatively simple to do.

Reply:
Okay will add to here and other places, originally didn't add as there is no memory being allocated.

Reply:
Memory is being allocated, but the compile will likely reuse some.